src/utils.ts

```diff
@@ -0,0 +1,67 @@
+import * as core from "@actions/core";
+import { read } from "@1password/op-js";
+import {
+	authErr,
+	envConnectHost,
+	envConnectToken,
+	envServiceAccountToken,
+	envManagedVariables,
+} from "./constants";
+
+export const semverToInt = (input: string): string =>
+	input
+		.split(".")
+		.map((n) => n.padStart(2, "0"))
+		.join("");
+
+export const validateAuth = (): void => {
+	let authType = "Connect";
+	if (!process.env[envConnectHost] || !process.env[envConnectToken]) {
+		if (!process.env[envServiceAccountToken]) {
+			throw new Error(authErr);
+		}
+		authType = "Service account";
+	}
+
+	// Adjust Connect host to have a protocol
+	if (
+		process.env[envConnectHost] &&
+		/* eslint-disable no-restricted-syntax */
+		!process.env[envConnectHost].startsWith("http://") &&
+		!process.env[envConnectHost].startsWith("https://")
+	) {
+		process.env[envConnectHost] = `http://${process.env[envConnectHost]}`;
+	}
+
+	core.debug(`Authenticated with ${authType}.`);
+};
+
+export const extractSecret = (
+	envName: string,
+	shouldExportEnv: boolean,
+): void => {
+	core.debug(`Populating variable: ${envName}`);
+	const ref = process.env[envName];
+	if (ref) {
+		const secretValue = read.parse(ref);
+		if (secretValue) {
+			if (shouldExportEnv) {
```

Comment:
I understand the chain of logic here, but nested `if` statements make my brain hurt :/ We have three in a row here: `if (ref)`, `if (secretValue)`, and `if (shouldExportEnv)`.

Is there a way that we can simplify this code by avoiding all of these nested `if`s?

Reply:
What I did here as a safe approach is to mirror the behavior in the shell script in Typescript. Now that we're confident it was properly migrated, we can simplify it. Also, I've realized that it became like this due to the linters complaining about `undefined` and such. 

Addressed in [`7c957c5`](https://github.com/1Password/load-secrets-action/pull/36/commits/7c957c570233d0664f9b7e566485c77163b11108).